beetmoverscript/utils.py

```diff
@@ -134,18 +147,26 @@ def update_props(props, platform_mapping):
     # l10n jobs have it set under `platform`. This is merely an uniformization
     # under the `stage_platform` field that is needed later on in the templates
     stage_platform = props.get("stage_platform", props.get("platform"))
+    # XXX Bug 1424482 - until we solve this, we need this hack. Since en-US
+    # have at least `stage_platform`, there is a way to tell whether they are
+    # devedition related or not. But for l10n jobs, we only have `platform`
+    # which is identical to the ones we have for Firefox.
```

Comment:
As mentioned in person, we can use graph generation to take `attributes['shipping_product']` and turn it into a `task.payload.product`; similarly we could use `build_platform` and turn it into a `task.payload.platform`. If we accept those as optional task payload schema items, and keyed off of them if they exist, then we could slowly reduce the need for `balrog_props.json`.